pkg/servicemesh/memberroll_test.go

```diff
@@ -0,0 +1,369 @@
+		} else {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+		}
+	}
+}
+
+func TestMemberRollCreate(t *testing.T) {
+	testCases := []struct {
+		testMemberRoll *MemberRollBuilder
+		expectedError  string
+	}{
+		{
+			testMemberRoll: buildValidMemberRollBuilder(buildMemberRollClientWithDummyObject()),
+			expectedError:  "",
+		},
+		{
+			testMemberRoll: buildInValidMemberRollBuilder(buildMemberRollClientWithDummyObject()),
+			expectedError:  "serviceMeshMemberRoll 'name' cannot be empty",
+		},
+		{
+			testMemberRoll: buildValidMemberRollBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:  "resourceVersion can not be set for Create requests",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testMemberRollBuilder, err := testCase.testMemberRoll.Create()
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, testMemberRollBuilder.Definition, testMemberRollBuilder.Object)
+		} else {
+			assert.Equal(t, testCase.expectedError, err.Error())
+		}
+	}
+}
+
+func TestMemberRollDelete(t *testing.T) {
+	testCases := []struct {
+		testMemberRoll *MemberRollBuilder
+		expectedError  error
+	}{
+		{
+			testMemberRoll: buildValidMemberRollBuilder(buildMemberRollClientWithDummyObject()),
+			expectedError:  nil,
+		},
+		{
+			testMemberRoll: buildInValidMemberRollBuilder(buildMemberRollClientWithDummyObject()),
+			expectedError:  fmt.Errorf("serviceMeshMemberRoll 'name' cannot be empty"),
+		},
+		{
+			testMemberRoll: buildValidMemberRollBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:  nil,
+		},
+	}
+
+	for _, testCase := range testCases {
+		err := testCase.testMemberRoll.Delete()
+
+		if testCase.expectedError == nil {
+			assert.Nil(t, testCase.testMemberRoll.Object)
+		} else {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+		}
+	}
+}
+
+func TestMemberRollUpdate(t *testing.T) {
+	testCases := []struct {
+		testMemberRoll *MemberRollBuilder
+		expectedError  string
+		members        []string
+	}{
+		{
+			testMemberRoll: buildValidMemberRollBuilder(buildMemberRollClientWithDummyObject()),
+			expectedError:  "",
+			members:        newMembersList,
+		},
+		{
+			testMemberRoll: buildInValidMemberRollBuilder(buildMemberRollClientWithDummyObject()),
+			expectedError:  "serviceMeshMemberRoll 'name' cannot be empty",
+			members:        newMembersList,
+		},
+	}
+
+	for _, testCase := range testCases {
+		assert.Equal(t, defaultMembersList, testCase.testMemberRoll.Definition.Spec.Members)
+		assert.Nil(t, nil, testCase.testMemberRoll.Object)
+		testCase.testMemberRoll.WithMembersList(testCase.members)
+		_, err := testCase.testMemberRoll.Update(true)
+
+		if testCase.expectedError != "" {
+			assert.Equal(t, testCase.expectedError, err.Error())
+		} else {
+			assert.Equal(t, testCase.members,
+				testCase.testMemberRoll.Definition.Spec.Members)
+		}
+	}
+}
+
+func TestMemberRollWithMembers(t *testing.T) {
```

Comment:
This function tests > WithMembersList. Can you re-name it to TestMemberRollWithMembersList

Reply:
Can you please re-name function?